options/components/webBottomContent/WebBottomContent.jsx

```diff
@@ -36,8 +37,21 @@ function WebUrlNewList() {
               className="mr-2 inline-block w-3 h-3"
               src={faviconURL(url.url)}
             />
-            {url.title}
-            {hasSearchResult ? <HighlightKeyword url={url} /> : ""}
+            {hasSearchResult &&
+            keyword &&
+            userSelectSearchValue === "키워드 검색" ? (
+              <HighlightKeyword url={url} />
+            ) : hasSearchResult &&
+              keyword &&
+              userSelectSearchValue === "제목 검색" ? (
+              <HighlightTitleKeyword url={url} />
+            ) : hasSearchResult &&
+              keyword &&
+              userSelectSearchValue === "제목 + 키워드 검색" ? (
+              <HighlightKeyword url={url} />
+            ) : (
+              url.title
+            )}
```

Comment:
반복되는 AND연산자와 삼항연사자의 조합이 가독성에 안좋은 영향을 끼치는 것 같습니다.
이 부분은 추후 리팩토링 작업 진행 시에 개선이 좀 필요해 보입니다.
지금 당장 개선할 수 있는 부분으로는 다음과 같이 hasSearchResult와 keyword의 유효성 검사가 계속 반복되지 않아도 해당 로직이 정상적으로 작동할 것이라 생각합니다.
```suggestion
            {hasSearchResult &&
            keyword &&
            userSelectSearchValue === "키워드 검색" ? (
              <HighlightKeyword url={url} />
            ) : userSelectSearchValue === "제목 검색" ? (
              <HighlightTitleKeyword url={url} />
            ) : userSelectSearchValue === "제목 + 키워드 검색" ? (
              <HighlightKeyword url={url} />
            ) : (
              url.title
            )}
```

Reply:
저도 작성하면서 가독성을 저해하는 코드라고 느꼈으나 구현 역량의 한계로 위와같은 코드를 작성하게 되었습니다.
좋은 지적에 감사드리며 전달 주신 부분 로직 참고하여 적용해 보았으나 제목만 입력 할 경우 제목의 단어하나하나가 map이 돌려져 뿌려지는 현상이 발생하여 적용하지 못하였습니다. 해당 코드 제안주시기 이전에 리팩토링이 필요한 코드라고 이야기주셨고 저 또한 이에 공감하기에 추후 리팩토링 기간에 어떻게 하면 가독성 좋은 코드로 개선 할 수 있을 지에 대한 연구를 해볼 수 있도록 하겠습니다.

Reply:
네 조건 횟수를 줄일 수도 있어보이고, 삼항연산자를 쓰지 않고도 할 수 있는 방법으로 할 수 있을것 같다고 생각됩니다. 그래서 다양한 방법으로 연구해보시면 좋을 것 같습니다. 추후 해당 코드가 어떻게 개선될지 기대가 됩니다.